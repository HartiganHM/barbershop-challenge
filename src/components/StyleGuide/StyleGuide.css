@import url("https://fonts.googleapis.com/css?family=Open+Sans");
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
.StyleGuide {
  align-self: center;
  display: flex;
  flex-direction: column;
  align-items: center; }
  .StyleGuide .home-link {
    display: flex;
    align-items: center;
    justify-content: space-around;
    height: 4rem;
    padding: 0.5rem;
    border: none;
    border-radius: 2px;
    background-color: #fff;
    color: #2F80ED;
    font-size: 1.6rem;
    transition: 0.2s ease-in-out;
    cursor: pointer;
    background-color: #2F80ED;
    color: #fff;
    cursor: default;
    display: flex;
    justify-content: center;
    width: 18rem;
    margin: 2rem;
    cursor: pointer;
    text-decoration: none; }
    .StyleGuide .home-link .home-icon {
      width: 16px;
      margin: 0.5rem;
      fill: rgba(255, 255, 255, 0.7);
      transition: 0.2s; }
    .StyleGuide .home-link:hover {
      background-color: #33ADFF;
      color: #fff; }
      .StyleGuide .home-link:hover .home-icon {
        fill: #fff;
        transform: rotate(-90deg); }
    .StyleGuide .home-link:active {
      background-color: #3F67D4;
      color: #fff; }
  .StyleGuide .page-title {
    color: #2F80ED;
    font-size: 3.2rem;
    text-align: center; }
  .StyleGuide .button-wrapper {
    display: flex;
    justify-content: space-between; }
    .StyleGuide .button-wrapper .button {
      display: flex;
      align-items: center;
      justify-content: space-around;
      height: 4rem;
      padding: 0.5rem;
      border: none;
      border-radius: 2px;
      background-color: #fff;
      color: #2F80ED;
      font-size: 1.6rem;
      transition: 0.2s ease-in-out;
      cursor: pointer;
      background-color: #2F80ED;
      color: #fff;
      cursor: default;
      display: flex;
      justify-content: center;
      width: 18rem;
      margin: 1.35rem;
      cursor: pointer; }
      .StyleGuide .button-wrapper .button .plus-icon {
        width: 16px;
        margin: 0.5rem;
        fill: rgba(255, 255, 255, 0.7);
        transition: 0.2s; }
      .StyleGuide .button-wrapper .button .loading-icon {
        display: inline-block;
        position: relative;
        width: 2.6rem;
        height: 2.6rem;
        align-self: center; }
        .StyleGuide .button-wrapper .button .loading-icon div {
          box-sizing: border-box;
          display: block;
          position: absolute;
          border-radius: 50%;
          animation: loader 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite; }
        .StyleGuide .button-wrapper .button .loading-icon div:nth-child(1) {
          animation-delay: -0.45s; }
        .StyleGuide .button-wrapper .button .loading-icon div:nth-child(2) {
          animation-delay: -0.3s; }
        .StyleGuide .button-wrapper .button .loading-icon div:nth-child(3) {
          animation-delay: -0.15s; }

@keyframes loader {
  0% {
    transform: rotate(0deg); }
  100% {
    transform: rotate(360deg); } }
        .StyleGuide .button-wrapper .button .loading-icon div {
          width: 2.6rem;
          height: 2.6rem;
          border: 3px solid #fff;
          border-color: #fff transparent transparent transparent; }
      .StyleGuide .button-wrapper .button.text-icon:hover .plus-icon {
        fill: #fff;
        transform: rotate(-90deg); }
      .StyleGuide .button-wrapper .button.hover-focus {
        background-color: #33ADFF;
        color: #fff; }
      .StyleGuide .button-wrapper .button.active {
        background-color: #3F67D4;
        color: #fff; }
      .StyleGuide .button-wrapper .button.disabled {
        background-color: #8EB7ED;
        color: #fff;
        pointer-events: none;
        cursor: not-allowed; }
      .StyleGuide .button-wrapper .button.loader {
        background-color: #8EB7ED;
        color: #fff;
        pointer-events: none;
        cursor: wait; }
    .StyleGuide .button-wrapper .button-text {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 18rem;
      margin: 1.35rem;
      padding: 0.5rem;
      color: #696969;
      font-size: 1.6rem; }

@media screen and (max-width: 620px) {
  .StyleGuide .page-title {
    font-size: 24px;
    width: 30rem; }
  .StyleGuide .button-wrapper {
    flex-direction: column; } }
