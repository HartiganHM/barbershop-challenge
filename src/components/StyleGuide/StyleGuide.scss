@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';
.StyleGuide {
  align-self: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  .home-link {
    @include buttonDefaultAlternate;
    width: 18rem;
    margin: 2rem;
    cursor: pointer;
    .home-wrapper {
      display: flex;
      height: 3rem;
      align-items: center;
      text-decoration: none;
      color: #fff;
      .home-icon {
        width: 16px;
        margin: 0.5rem;
        fill: $white-transparent;
        transition: 0.2s;
      }
      .home-text {
        align-self: baseline;
        margin: 0.75rem 0.5rem;
        padding: 0;
        border: none;
        background: none;
        color: #fff;
        cursor: pointer;
      }
    }
    &:hover {
      @include buttonHover;
      .home-icon {
        fill: #fff;
        transform: rotate(-90deg);
      }
    }
    &:active {
      @include buttonActive;
    }
  }
  .page-title {
    color: $blue-midtone;
    font-size: 3.2rem;
    text-align: center;
  }
  .button-wrapper {
    display: flex;
    justify-content: space-between;
    .button {
      @include buttonDefaultAlternate;
      display: flex;
      justify-content: center;
      width: 18rem;
      margin: 1.35rem;
      cursor: pointer;
      .wrapper {
        display: flex;
        height: 3rem;
        align-items: center;
        .plus-icon {
          width: 16px;
          margin: 0.5rem;
          fill: $white-transparent;
          transition: 0.2s;
        }
        .loading-icon {
          @include buttonLoader;
          align-self: center;
        }
        .get-photos-text {
          align-self: baseline;
          margin: 0.75rem 0.5rem;
          padding: 0;
          border: none;
          background: none;
          color: #fff;
          cursor: pointer;
        }
      }
      &.text-icon:hover {
        .plus-icon {
          fill: #fff;
          transform: rotate(-90deg);
        }
      }
      &.hover-focus {
        @include buttonHover;
      }
      &.active {
        @include buttonActive;
      }
      &.disabled {
        @include buttonDisabled;
        cursor: not-allowed;
      }
      &.loader {
        @include buttonDisabled;
        cursor: wait;
      }
    }
    .button-text {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 18rem;
      margin: 1.35rem;
      padding: 0.5rem;
      color: $darkgrey;
      font-size: 1.6rem;
    }
  }
}

@media screen and (max-width: 620px) {
  .StyleGuide {
    .page-title {
      font-size: 24px;
      width: 30rem;
    }
    .button-wrapper {
      flex-direction: column;
    }
  }
}
