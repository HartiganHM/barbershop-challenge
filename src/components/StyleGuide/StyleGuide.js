import React from 'react';
import { Link } from 'react-router-dom';
import generateUniqueKey from '../../helpers/generateUniqueKey/generateUniqueKey';
import iconData from '../../data/iconData';
import './StyleGuide.css';

const StyleGuide = () => {
  const buttons = [
    'Text + Icon',
    'Text',
    'Hover / Focus',
    'Active',
    'Disabled',
    'Loader'
  ];
  const buttonClasses = [
    'text-icon',
    'text',
    'hover-focus',
    'active',
    'disabled',
    'loader'
  ];

  const plusIcon = (
    <svg
      key={generateUniqueKey()}
      viewBox="0 0 13.33 13.33"
      className="plus-icon"
    >
      <path d={iconData.getPhotos} />
    </svg>
  );

  const loader = (
    <div className="loading-icon">
      <div />
      <div />
      <div />
      <div />
    </div>
  );

  const renderedButtons = buttons.map((button, index) => {
    const buttonClass = 'button ' + buttonClasses[index];
    return (
      <span key={generateUniqueKey()} className="button-wrapper">
        <button className={buttonClass}>
          <a className="wrapper">
            {button === 'Text + Icon' && plusIcon}
            {button === 'Loader' && loader}
            {button !== 'Loader' && (
              <input className="get-photos-text" type="submit" value="Click Me" />
            )}
          </a>
        </button>

        <h4 className="button-text">{button}</h4>
      </span>
    );
  });

  return (
    <div className="StyleGuide">
      <button className="home-link">
        <Link to="/" className="home-wrapper">
          <svg viewBox="0 0 612 612" className="home-icon">
            <path d={iconData.home} />
          </svg>
          <input className="home-text" type="submit" value="Home" />
        </Link>
      </button>

      <h2 className="page-title">Barbershop Challenge Style Guide</h2>

      <div>{renderedButtons}</div>
    </div>
  );
};

export default StyleGuide;
