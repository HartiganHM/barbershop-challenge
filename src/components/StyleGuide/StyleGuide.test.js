import React from 'react';
import { shallow } from 'enzyme';
import StyleGuide from './StyleGuide';

describe('StyleGuide tests', () => {
  it('Should match the snapshot', () => {
    const renderedStyleGuide = shallow(<StyleGuide />);

    expect(renderedStyleGuide).toMatchSnapshot();
  });
});
