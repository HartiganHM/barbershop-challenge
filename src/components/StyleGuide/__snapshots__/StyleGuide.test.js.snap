// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleGuide tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyleGuide />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Soi una StyleGuide
      </h1>,
      "className": "StyleGuide",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Soi una StyleGuide",
      },
      "ref": null,
      "rendered": "Soi una StyleGuide",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Soi una StyleGuide
        </h1>,
        "className": "StyleGuide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Soi una StyleGuide",
        },
        "ref": null,
        "rendered": "Soi una StyleGuide",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
